flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -95,15 +100,55 @@ public static ClassLoader buildUserCodeClassLoader(List<URL> jars, List<URL> cla
 		return FlinkUserCodeClassLoaders.parentFirst(urls, parent);
 	}
 
+	public static JobExecutionResult submitJob(
+			ClusterClient<?> client,
+			JobGraph jobGraph) throws ProgramInvocationException {
+		try {
+			return client
+				.submitJob(jobGraph)
+				.thenApply(JobSubmissionResult::getJobID)
+				.thenApply(DetachedJobExecutionResult::new)
+				.get();
+		} catch (InterruptedException | ExecutionException e) {
+			ExceptionUtils.checkInterrupted(e);
+			throw new ProgramInvocationException("Could not run job in detached mode.", jobGraph.getJobID(), e);
+		}
+	}
+
+	public static JobExecutionResult submitJobAndWaitForResult(
+			ClusterClient<?> client,
+			JobGraph jobGraph,
+			ClassLoader classLoader) throws ProgramInvocationException {
+		JobResult jobResult;
+
```

Comment:
You can add the necessary precondition checks like `checkNotNull()` for the `client`, the `jobGraph` and the `classLoader`.